repobuilder/job.go

```diff
@@ -524,21 +524,38 @@ func (j *repoBuilderJob) processPackages(ctx context.Context) error {
 		}
 
 		var expandedPath string
-		for archiverName, ff := range archiver.SupportedFormats {
-			if !ff.Match(localPath) {
-				continue
-			}
-
-			expandedPath = filepath.Join(j.tmpdir, fmt.Sprintf("extracted-%d-%s", idx, archiverName))
-			if err := ff.Open(localPath, expandedPath); err != nil {
-				catcher.Add(err)
-			}
-
+		format, err := archiver.ByExtension(localPath)
+		if err != nil {
+			catcher.Wrapf(err, "unrecognized archive extension for file '%s'", localPath)
 			break
 		}
-		if expandedPath == "" {
-			catcher.Errorf("could not expand archive for %s", localPath)
+		unarchiveFormat, ok := format.(archiver.Unarchiver)
+		if !ok {
+			catcher.Errorf("file '%s' does not have any known means to unarchive it", localPath)
+			break
 		}
+		// kim: QUESTION: does this depend on using the archiver name?
```

Comment:
Should we clean up these comments?

Reply:
Yeah, these were old thoughts. I was thinking that it would be preferable to not change the repobuilder jobs because they're so untested, but I don't think it's worth keeping the old archiver version and newer one at the same time.